### ISSUE TITLE
Discord Plugin

### ISSUE THREAD

**Issue by nerdtron123** [keybase/keybase-issues#2991]
You should make a discord plugin for this.

**Comment by tkdberger** [comment]
I think it's a matter of "where can you post it". Every service keybase currently allows someone to prove an identity on has a way to easily link to the proof (allowing other people to verify the identity claim). Discord lacks message permalinks, or any way to link to a specific message at all. This makes Discord proofs just about impossible while still being publicly verifiable.

**Comment by zQueal** [comment]
> Discord lacks message permalinks, or any way to link to a specific message at all.

Yes and no. Each message has a unique debugging identifier which can be used to pull a message via the Discord API. So yes it's possible to do programatically, but not publicly. So yes, Discord isn't a very ideal platform for proofs.

**Comment by nerdtron123** [comment]
there's a search feature though

**Comment by skwerlman** [comment]
the search is limited to a certain number of messages, so it's not sustainable as a solution

**Comment by zQueal** [comment]
Not only is it restricted to a relatively small number of entries it becomes increasingly cumbersome for someone to actually find a proof unless you're using text objects like hashtags.

Using the Discord Webhook API it makes it relatively simple to hook into a Discord text channel to "search" for proofs if you use something like hashtags, but after a while, considering the size of Keybases' user base it would be impossible to continue public proof verification--which IMO defeats the entire purpose.

Regardless, I've added it to #518 feel free to close this issue.

**Comment by comp500** [comment]
Discord now supports message permalinks, by pressing Copy Link on a message: 
![image](https://user-images.githubusercontent.com/1138417/57571496-050f6980-7407-11e9-9fdf-ed2a7a8a9ebd.png)
That produces a link that looks like https://discordapp.com/channels/176780432371744769/191289558329982976/570430574985216022 that can be clicked in Discord to jump to the message.
Another issue, however, is that Discord usernames and numbers can be changed at any time, so the unique ID (Twitter snowflake) of the user needs to be used, at least in the backend. This can be obtained in the app by enabling Developer Mode (User Settings -> Appearance) then right clicking the user's avatar.
